sensors/sqs_sensor.py

```diff
@@ -98,50 +111,117 @@ def _may_setup_sqs(self):
         else:
             self.input_queues = []
 
-        self.aws_access_key = self._get_config_entry('aws_access_key_id')
-        self.aws_secret_key = self._get_config_entry('aws_secret_access_key')
-        self.aws_region = self._get_config_entry('region')
+        # checker configuration is update, or not
+        def _is_same_credentials(session, account_id):
+            c = session.get_credentials()
+            same_credentials = c is not None and \
+                c.access_key == self.credentials[account_id][0] and \
+                c.secret_key == self.credentials[account_id][1]
+            if account_id != self.account_id:
+                return same_credentials and c.token == self.credentials[account_id][2]
+            else:
+                return same_credentials
```

Comment:
Either refactor this as successive ifs, or keep it a boolean expression, do not combine them (add `and (account_id == self.account_id or c.token == self.credentials[account_id][2])`